sdk/identity/identity/review/identity.api.md

```diff
@@ -287,6 +288,17 @@ export interface ErrorResponse {
     traceId?: string;
 }
 
+// @public
+export function getBearerTokenProvider(credential: TokenCredential, scopes: string | string[], options?: GetBearerTokenProviderOptions): () => Promise<string>;
```

Comment:
Last night I got to thinking - our credentials are usually classes that get instantiated. `new MyCoolCredential(options)`

I'm wondering if we should stay consistent here. on the one hand, this is a factory function in an ocean of classes. On the other hand, it's not like this is an actual TokenCredential - it's a wrapper around a credential right?

...

But on the 47th hand 😄 EnvironmentCredential / ChainedTokenCredential / DefaultAzureCredential are all "wrappers" around other credentials and maintain the class-based interface.

So, should this be a class? If not, why not? I have no strong opinions but wanted to bring it up for discussion